app/controllers/application_controller.rb

```diff
@@ -1517,29 +1464,11 @@ def show_index_of_objects(query, args = {})
 
     # Otherwise paginate results.  (Everything we need should be cached now.)
     else
-      @pages = if args[:letters]
-                 paginate_letters(letter_arg, number_arg, num_per_page)
-               else
-                 paginate_numbers(number_arg, num_per_page)
-               end
-
-      # Skip to correct place if coming back in to index from show_object.
-      if args[:id].present? &&
-         params[@pages.letter_arg].blank? &&
-         params[@pages.number_arg].blank?
-        @pages.show_index(query.index(args[:id]))
-      end
-
-      # Instantiate correct subset.
-      logger.warn("QUERY starting: #{query.query.inspect}")
-      @timer_start = Time.current
-      @objects = query.paginate(@pages, include: include)
-      @timer_end = Time.current
-      logger.warn("QUERY finished: model=#{query.model}, " \
-                  "flavor=#{query.flavor}, params=#{query.params.inspect}, " \
-                  "time=#{(@timer_end - @timer_start).to_f}")
+      calc_pages(args, query)
+      find_objects(args, query)
 
       # Give the caller the opportunity to add extra columns.
+      #### Refactor problem?????
```

Comment:
Do we really need to be able to pass a block to this function?  Makes refactoring a lot harder and I was only able to find one seemingly obscure usage of this functionality.

Reply:
The only place it's used is the old /name/authored_names (≡ /names?with_descriptions=1).

My personal opinion is basically that descriptions are so rarely used, contribute so little to our core mission, and contribute so much to our tech debt, that I encourage cutting back on that functionality wherever possible.